employees/views.py

```diff
@@ -162,69 +179,73 @@ def _create_project_join_form(self) -> ProjectJoinForm:
         self.hide_join = not project_form_queryset.exists()
         return ProjectJoinForm(queryset=project_form_queryset)
 
-    def initial(self, request: HttpRequest, *args: Any, **kwargs: Any) -> None:
-        logger.info(f"Initial method for user with id: {self.request.user.pk} in ReportList view")
-        super().initial(request, *args, **kwargs)
-        self.reports = self.get_queryset()
-        self.daily_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_dates()
-        self.monthly_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_authors()
-
-    def get(self, _request: HttpRequest) -> Response:
-        logger.info(f"User with id: {self.request.user.pk} get to the ReportList view")
-        return Response(
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context_data = super().get_context_data(**kwargs)
+        year = int(kwargs["year"])
+        month = int(kwargs["month"])
+        context_data.update(
             {
-                "serializer": self._create_serializer(),
+                "serializer": self._create_serializer(self._default_date(year, month)),
                 "object_list": self.reports,
                 "daily_hours_sum": self.daily_hours_sum,
                 "monthly_hours_sum": self.monthly_hours_sum,
                 "UI_text": ReportListStrings,
                 "project_form": self._create_project_join_form(),
                 "hide_join": self.hide_join,
+                "post_date": [year, month],
             }
         )
+        return context_data
 
-    def post(self, request: HttpRequest) -> Response:
-        logger.info(f"User with id: {request.user.pk} sent post to the ReportList view")
-        reports_serializer = self._create_serializer(data=request.data)
-        if "join" in request.POST:
+    def initial(self, request: HttpRequest, *args: Any, **kwargs: Any) -> None:
+        logger.info(f"Initial method for user with id: {self.request.user.pk} in ReportList view")
+        super().initial(request, *args, **kwargs)
+        self.reports = self.get_queryset()
+        self.daily_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_dates()
+        self.monthly_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_authors()
+        self.reports = self.get_queryset()
+        self.context_data = self.get_context_data(**kwargs)
+
+    def get(self, _request: HttpRequest, year: int, month: int) -> Response:  # pylint: disable=unused-argument
+        return Response(self.context_data)
+
+    def post(  # type: ignore
+        self, request: HttpRequest, year: int, month: int
+    ) -> Union[Response, HttpResponseRedirectBase]:
+        year = int(year)
```

Comment:
this seems awkward - `year` and `month` are already declared as  `int`s in the function's signature. Why should we be casting them?

Reply:
I just checked this and the parameters provided are indeed in string format. This causes an error as soon as `_default_date()` is called. I do wonder myself why there are no warnings raised regarding integer typing, if in reality a string is returned. 
![Zrzut ekranu 2019-05-29 o 16 45 26](https://user-images.githubusercontent.com/17569113/58569457-6cfee600-8236-11e9-9cab-ed56aca54dcf.png)

Reply:
For url resolving Python regex lib is used, so the matched object is string.